pkg/kubelet/cm/dra/plugin/noderesources.go

```diff
@@ -259,9 +260,20 @@ func (c *nodeResourcesController) run(ctx context.Context) {
 	// because eventually these list calls succeed.
 	// TODO (https://github.com/kubernetes/kubernetes/issues/123691): can we avoid these error log entries? Perhaps wait here?
 
+	var node *v1.Node
+	var err error
+	for {
+		node, err = c.getNode()
```

Comment:
Could it be that this node changes later? I don't know, therefore I am wary of making the assumption that it doesn't.

Perhaps better call this whenever an object needs to be updated or created? It's reading from a local cache, so it's not that expensive.

I don't think we need to re-sync when the node object changes. The only way how this can cause the UID to change is when the old node object gets removed. Unless users work really hard to breaks us, all old slices will get removed, which triggers syncing.

Reply:
Thank you for the review and suggestions!

I agree that node name or UID are unlikely to change. However, `getNode` will always get the latest available node object version from the cache. At least I hope it will. Alternatively we can just store node name and UID in the `nodeResourcesController struct`. What would you suggest?

Reply:
I suggest to call `getNode` anew in `sync`

Reply:
I missed that because this code here remained.

Can you simplify it (no need for separate variables, no need for `node`) and add a comment why we have this loop here?

Reply:
ok, will do. Thanks for the explanations :)